From eb263853ccaa02906a31caec2accd2b3573e832e Mon Sep 17 00:00:00 2001
From: Rob Herring <r.herring@freescale.com>
Date: Fri, 5 Jun 2009 10:58:43 +0800
Subject: [PATCH] ENGR00112775 ipu: fix active line display timings

The active line timer offset was not setup correctly to use
vsync width and upper margin.

Signed-off-by: Rob Herring <r.herring@freescale.com>
---
 drivers/mxc/ipu3/ipu_disp.c         |    2 +-
 drivers/video/mxc/mxc_ipuv3_fb.c    |    4 +---
 drivers/video/mxc/mxcfb_claa_wvga.c |    2 +-
 3 files changed, 3 insertions(+), 5 deletions(-)

diff --git a/drivers/mxc/ipu3/ipu_disp.c b/drivers/mxc/ipu3/ipu_disp.c
index b8a28c9..a133585 100644
--- a/drivers/mxc/ipu3/ipu_disp.c
+++ b/drivers/mxc/ipu3/ipu_disp.c
@@ -1085,7 +1085,7 @@ int32_t ipu_init_sync_panel(int disp, uint32_t pixel_clk,
 
 		/* Setup active data waveform to sync with DC */
 		_ipu_di_sync_config(disp, 4, 0, DI_SYNC_HSYNC,
-				    v_start_width, DI_SYNC_HSYNC, height,
+				    v_sync_width + v_start_width, DI_SYNC_HSYNC, height,
 				    DI_SYNC_VSYNC, 0, DI_SYNC_NONE,
 				    DI_SYNC_NONE, 0, 0);
 		_ipu_di_sync_config(disp, 5, 0, DI_SYNC_CLK,
diff --git a/drivers/video/mxc/mxc_ipuv3_fb.c b/drivers/video/mxc/mxc_ipuv3_fb.c
index be744ed..336cb1a 100644
--- a/drivers/video/mxc/mxc_ipuv3_fb.c
+++ b/drivers/video/mxc/mxc_ipuv3_fb.c
@@ -141,7 +141,6 @@ static int mxcfb_set_par(struct fb_info *fbi)
 	ipu_di_signal_cfg_t sig_cfg;
 	ipu_channel_params_t params;
 	struct mxcfb_info *mxc_fbi = (struct mxcfb_info *)fbi->par;
-	uint32_t v_to_h_sync;
 
 	dev_dbg(fbi->device, "Reconfiguring framebuffer\n");
 
@@ -216,7 +215,6 @@ static int mxcfb_set_par(struct fb_info *fbi)
 				       fbi->var.vsync_len,
 				       fbi->var.lower_margin, sig_cfg) != 0) {
 #else
-		v_to_h_sync = (mxc_fbi->ipu_di) ? 0 : 480;
 		if (ipu_init_sync_panel(mxc_fbi->ipu_di,
 					(PICOS2KHZ(fbi->var.pixclock)) * 1000UL,
 					fbi->var.xres, fbi->var.yres,
@@ -227,7 +225,7 @@ static int mxcfb_set_par(struct fb_info *fbi)
 					fbi->var.upper_margin,
 					fbi->var.vsync_len,
 					fbi->var.lower_margin,
-+					v_to_h_sync, sig_cfg) != 0) {
+					0, sig_cfg) != 0) {
 #endif
 			dev_err(fbi->device,
 				"mxcfb: Error initializing panel.\n");
diff --git a/drivers/video/mxc/mxcfb_claa_wvga.c b/drivers/video/mxc/mxcfb_claa_wvga.c
index af0e069..2e03231 100644
--- a/drivers/video/mxc/mxcfb_claa_wvga.c
+++ b/drivers/video/mxc/mxcfb_claa_wvga.c
@@ -48,7 +48,7 @@ static int lcd_on;
 static struct fb_videomode video_modes[] = {
 	{
 	 /* 800x480 @ 55 Hz , pixel clk @ 25MHz */
-	 "CLAA-WVGA", 55, 800, 480, 40000, 80, 0, 10, 0, 20, 10,
+	 "CLAA-WVGA", 55, 800, 480, 40000, 40, 40, 5, 5, 20, 10,
 	 FB_SYNC_CLK_LAT_FALL,
 	 FB_VMODE_NONINTERLACED,
 	 0,},
-- 
1.5.4.4

